src/bindings/python/src/pyopenvino/utils/utils.cpp

```diff
@@ -253,7 +253,34 @@ py::object from_ov_any(const ov::Any& any) {
 std::map<std::string, ov::Any> properties_to_any_map(const std::map<std::string, py::object>& properties) {
     std::map<std::string, ov::Any> properties_to_cpp;
     for (const auto& property : properties) {
-        properties_to_cpp[property.first] = Common::utils::py_object_to_any(property.second);
+        if (property.first == ov::cache_crypto_callback.name()) {
+            auto property_value = property.second;
+            std::function<std::string(const std::string&)> encrypt_func =
+                [property_value](const std::string& in_str) -> std::string {
+                // Acquire GIL, execute Python function
+                py::gil_scoped_acquire acquire;
+                if (!py::isinstance<py::list>(property_value)) {
+                    OPENVINO_THROW("The value type of ov::cache_crypto_callback property is expected list");
```

Comment:
we can merge condition if and throw to openvino_assert

the same below

Reply:
updated